src/pyasm/command/csv_import_cmd.py

```diff
@@ -246,21 +265,22 @@ def execute(my):
 
             
             try:
-                #sobject.commit(triggers=False)
                 sobject.commit(triggers=my.triggers_mode)
 
                 if note:
                     note_obj = SearchType.create("sthpw/note")
                     note_obj.set_value("note", note)
-                    note_obj.set_value("process", "publish")
-                    note_obj.set_value("context", "publish")
+                    note_process = my.note_processes[i]
+                    if not note_process:
+                        note_process = "publish"
+                    note_obj.set_value("process", note_process)
+                    note_obj.set_value("context", note_process)
                     note_obj.set_user()
                     note_obj.set_parent(sobject)
                     note_obj.commit()
 
             except SqlException, e:
                 msg = "Error creating new entry for row [%s]: %s, %s" % (row_count, str(row), e.__str__() )
```

Comment:
so this check() function is also raising Exception? Is there a Dry run function there?